### PR TITLE
Multi Tanent Application 

### PR DESCRIPTION
**Pull request by cham11ng**
Different Schemas
Middleware
Common Migration Foreign Key Assignment
Helper Function for Dynamic Subdomain routing.

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class HomeController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     *
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
```
Comment (cham11ng) [inline app/Http/Controllers/HomeController.php:13]: This `auth` middleware execute first before `select-schema`. This is main issue.

File: resources/views/auth/login.blade.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
+@extends('layouts.app')
+
+@section('content')
+<div class="container">
+    <div class="row">
+        <div class="col-md-8 col-md-offset-2">
+            <div class="panel panel-default">
+                <div class="panel-heading">Login</div>
+                <div class="panel-body">
+                    <form class="form-horizontal" role="form" method="POST" action="{{ route('login', ['tenant' => $tenant]) }}">
```
Comment (cham11ng) [inline resources/views/auth/login.blade.php]:
```php
{{ route('login', ['tenant' => $tenant]) }}
```
everywhere where we use `route('login')`we have to pass $tenant. Is there better solution?

File: app/Http/Controllers/Auth/LoginController.php [hunk 3]
```diff
@@ -30,10 +35,30 @@ class LoginController extends Controller
...
+
+    /**
+     * Show the application's login form.
+     *
+     * @param $tenant
+     * @return \Illuminate\Http\Response
+     */
+    public function showLoginForm($tenant)
+    {
+        return view('auth.login', compact('tenant'));
```
Comment (cham11ng) [inline app/Http/Controllers/Auth/LoginController.php]: we have to pass `$tenant` as well that comes from `{tenant}` wildcard

**Review by cham11ng — COMMENTED** [review]
Changes Reviews

File: resources/views/auth/login.blade.php (hunk 2 above)
Comment (sachit-singh) [reply resources/views/auth/login.blade.php]: get tenant from subdomain..

Comment (cham11ng) [reply resources/views/auth/login.blade.php]: I've already done that, I just want a better solution to auto recognize this route name if possible.

File: app/Http/Controllers/HomeController.php (hunk 1 above)
Comment (cham11ng) [reply app/Http/Controllers/HomeController.php:13]: This was fixed by this commit https://github.com/cham11ng/multi-tenant-postgres/pull/1/commits/d4665da9eff9ac29c04896704a093f80d87ec255

**Comment by neharajbanshi** [comment]
looks okay

File: app/Tenant.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        'sub_domain',
+        'email'
+    ];
+
+    /**
+     * This table is in public schema
+     *
+     * @var string
+     */
+    protected $table = 'public.tenants';
```
Comment (neharajbanshi) [inline app/Tenant.php]: It should be dynamic

**Review by neharajbanshi — APPROVED** [review]

**Review by sachit-singh — APPROVED** [review]

File: app/Http/Controllers/Auth/ResetPasswordController.php [hunk 5]
```diff
@@ -30,10 +31,35 @@ class ResetPasswordController extends Controller
...
+    /**
+     * Display the password reset view for the given token.
+     *
+     * If no token is present, display the link request form.
+     *
+     * @param \Illuminate\Http\Request|Request $request
+     * @param  string|null                     $token
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function showResetForm(Request $request, $tenant, $token = null)
```
Comment (cham11ng) [inline app/Http/Controllers/Auth/ResetPasswordController.php:49]: I've to always catch `$tenant` here because of `{tenant}` wild card `web.php`.

**Pull request merged**